Source/GSXib5KeyedUnarchiver.m

```diff
@@ -3563,11 +3569,19 @@ - (BOOL) containsValueForKey: (NSString *)key
         {
           hasValue = [self containsValueForKey: [XmlKeyMapTable objectForKey: key]];
         }
+      else if ([@"NSBackgroundColors" isEqualToString: key])
+        {
+          hasValue = [currentElement elementForKey: @"primaryBackgroundColor"] != nil;
+        } 
+      else if ([@"NSCollectionViewLayout" isEqualToString: key])
```

Comment:
This case should work out of the box. Are you sure you need this code?

Reply:
When I remove this the layout is no longer rendered.  I'm not sure why.   I'm leaving this in for the time being.

Reply:
This is technically redundant (as is line 3328), as the name translation SHOULD take care of these.  

Of course, unless the processing sequence has been broken somewhere.

Reply:
The mapping method does the following:

![screenshot](https://user-images.githubusercontent.com/27943/204151491-24a028db-f05a-46a8-8eed-c0703e527adf.png)

So it is not a direct mapping.   I believe that, by default, it filters out the entire class name instead of just NS.  In this case we just need NS filtered out, thus the code.   Of course I think there is one other way to do this using the dictionary that is hard coded above.  I've been thinking about moving that into a file that is read as a resource of the bundle (a thought for another time and, likely, not a good idea).

Reply:
I'm closing this... reopen, if there is an objection to the above reasoning.

Reply:
I see the problem, as I have the original GNUstep code that I implemented here, and just compared it to the community version.

It turns out that when the community accepted the code it made some changes, particular to lines 

      else if ([key hasPrefix: @"NS"])
        {
          // Try a key minus a (potential) NS prefix...
          NSString *newKey = [self alternateName: key];
          hasValue = [self containsValueForKey: newKey];
        }

which originally was:

      else if ([key hasPrefix:@"NS"])
        {
          // Try a key minus a (potential) NS prefix...
          NSString *newKey = [key stringByDeletingPrefix:@"NS"];
          newKey = [[[newKey substringToIndex:1] lowercaseString] stringByAppendingString:[newKey substringFromIndex:1]];
          hasValue = [self containsValueForKey:newKey];
        }

All the original code was doing was remove the “NS” prefix, and accept the rest of the word as a key with the first letter lowercased.

The new code will not only remove the NS prefix but also anything associated with the potential key, the key in this case probably "collectionViewLayout".  The “alternateName” method changes the NSCollectionViewLayout to "", so I see what you’re saying.  That is, if I follow the code correctly.  Thus, your need to add an explicit entry translation.

I am not sure why that code section was changed, but I suspect it was either due to an attempt at optimization OR a misunderstanding of the original intent was for it.   I guess there could have been some valid reason but since I wasn't involved I can only guess.  It suspect it will be a source of confusion and contention in there going forward as I am not sure of even how many things it could have potentially have broken based on what I was doing, and thus all the new entries in that section.

Anyway, sorry for my original comment(s) on the code section as I hadn't followed through what code the community had actually decided to accept, thinking it was pretty much mostly what I had put together.  I just went through the community code and compared it against what I had originally done, thus finding this discrepancy.  I will try to avoid such confusion going forward if I decide to comment.

Reply:
No need for apologies... your input is always appreciated, of course.  I don't think it was due to a misunderstanding of the original intent.  A lot of the time (majority of cases) the key ends up being mapped from NSClassNameKeyName => keyName, so it was a sensible change.   As we all know, however, Apple sometimes breaks its own conventions.

Reply:
I understand why this isn't working, but we really should come up with a better solution. Either add this to the XmlKeyMapTable, which is always consulted first. Or we could change the method alternateName: to return a list of names that should be tried one after another.
And remember that all the special cases present here need also to be handled in containsValueForKey:.

Reply:
@fredkiefer Agreed.   I believe it should try removing "NS" first and then "NS{ClassName}" second.  I am not sure which is the most common case, but that does seem to be what's implied.  Let me know what you think.   I will implement this in this branch.  If you would like me to move it to a separate branch, I will do so.

Reply:
I opted to do this the simple way by adding it to the XmlKeyMapTable.  I will add the more "generic" logic in a separate branch.